substrate/frame/nomination-pools/src/tests.rs

```diff
@@ -3239,179 +3248,29 @@ mod pool_withdraw_unbonded {
 
 	#[test]
 	fn pool_withdraw_unbonded_works() {
-		ExtBuilder::default().build_and_execute(|| {
+		ExtBuilder::default().add_members(vec![(20, 10)]).build_and_execute(|| {
 			// Given 10 unbond'ed directly against the pool account
-			assert_ok!(StakingMock::unbond(&default_bonded_account(), 5));
-			// and the pool account only has 10 balance
-			assert_eq!(StakingMock::active_stake(&default_bonded_account()), Ok(5));
-			assert_eq!(StakingMock::total_stake(&default_bonded_account()), Ok(10));
-			assert_eq!(Balances::free_balance(&default_bonded_account()), 10);
+
+			assert_ok!(Pools::unbond(RuntimeOrigin::signed(20), 20, 5));
+
+			assert_eq!(StakingMock::active_stake(&default_bonded_account()), Ok(15));
+			assert_eq!(StakingMock::total_stake(&default_bonded_account()), Ok(20));
+			assert_eq!(Balances::free_balance(&default_bonded_account()), 20);
 
 			// When
+			CurrentEra::set(StakingMock::current_era() + StakingMock::bonding_duration() + 1);
 			assert_ok!(Pools::pool_withdraw_unbonded(RuntimeOrigin::signed(10), 1, 0));
 
 			// Then there unbonding balance is no longer locked
-			assert_eq!(StakingMock::active_stake(&default_bonded_account()), Ok(5));
-			assert_eq!(StakingMock::total_stake(&default_bonded_account()), Ok(5));
-			assert_eq!(Balances::free_balance(&default_bonded_account()), 10);
+			assert_eq!(StakingMock::active_stake(&default_bonded_account()), Ok(15));
+			assert_eq!(StakingMock::total_stake(&default_bonded_account()), Ok(15));
+			assert_eq!(Balances::free_balance(&default_bonded_account()), 20);
 		});
 	}
 }
 
 mod withdraw_unbonded {
 	use super::*;
-	use sp_runtime::bounded_btree_map;
-
-	#[test]
-	fn withdraw_unbonded_works_against_slashed_no_era_sub_pool() {
```

Comment:
Is this test not relevant anymore?

Reply:
No, I introduced it but realized it doesn't make sense. There will never be slashes against the no_era pool.

Reply:
Thats right. No idea how that happened. I will restore it.